src/config.js

```diff
@@ -1,21 +1,46 @@
 /* eslint-disable import/no-dynamic-require */
 /* eslint-disable global-require */
 const path = require('path')
+const joi = require('joi')
 
-if (!process.env.NODE_ENV) { process.env.NODE_ENV = 'development' }
+const nodeEnvSchema = joi.object({
+	  NODE_ENV: joi.string()
+	    .valid('development', 'production', 'test')
+		.default('development')
+}).unknown()
 
-if (process.env.NODE_ENV === 'development') { require('dotenv').config() }
-if (process.env.NODE_ENV === 'test') {
+const environmentSchema = joi.object().keys({	
+	DATABASE_URL: joi.string().required(),
+	PORT: joi.number().required(),
+	TOKEN_SECRET: joi.string().required(),
+	EMAIL_SERVICE_HOST: joi.string().required(),
+	EMAIL_SERVICE_PORT: joi.number().required(),
+	EMAIL_SERVICE_PASSWORD: joi.string().required(),
+	EMAIL_SERVICE_USERNAME: joi.string().required(),
+	EMAIL_FROM: joi.string().required(),
+	FRONTEND_BASE_URL: joi.string().required(),
+	ORGANIZATION_NAME: joi.string().required()
+}).unknown()
+
+const validateSchema = (schema, env) => {
+	const { error, value } = schema.prefs({
+		errors: { label: 'key' }
+	}).validate(env)
+	if (error) {
+		throw new Error(`Config validation error: ${error.message}`)
+	}
+	return value
+}
+
+const nodeEnv = validateSchema(nodeEnvSchema, process.env)
+
+if (nodeEnv.NODE_ENV === 'development') { require('dotenv').config() }
+if (nodeEnv.NODE_ENV === 'test') {
 	require('dotenv').config({ 
 		path: path.resolve( '.env.test') 
 	})}
 
-
 module.exports = (key) => {
-	const value = process.env[key]
-	if (value === undefined) {
-		throw new Error(`No config for env variable ${key}`)
-	} else {
-		return value
-	}
+	const value = validateSchema(environmentSchema, process.env)
```

Comment:
this is cool but we want the validation to happen outside the function so that the server will not even start. That is we want it throw an error right away when any of the required envs are not available